tests/system/python/packages/test_authentication.py

```diff
@@ -1322,21 +1284,54 @@ def test_on_custom_port(self, wait_time):
         # FIXME: Remove this wait time
         time.sleep(wait_time)
 
-        conn.request("PUT", '/fledge/restart', json.dumps({}))
+        # Restart Fledge
+        restart_headers = {}  # No auth headers needed for restart endpoint
+        conn.request("PUT", '/fledge/restart', headers=restart_headers, body=json.dumps({}))
         r = conn.getresponse()
         assert 200 == r.status
         r = r.read().decode()
         jdoc = json.loads(r)
         assert "Fledge restart has been scheduled." == jdoc['message']
-
-        time.sleep(wait_time * 2)
-
+        
+        # Note: Can't use restart_and_wait_for_fledge() from conftest because it only supports
+        # default ports and HTTP connections, but we need to verify HTTPS on custom port 2005
+        # Wait for Fledge to restart and verify it's running on the new port (2005)
+        print(f"Waiting for Fledge to restart on port 2005... (Initial wait: {wait_time}s)")
+        time.sleep(wait_time)
+        
+        start_time = time.time()
+        max_retries = 5
+        ping_headers = {}  # No auth headers needed for ping endpoint
```

Comment:
is not entirely accurate; it depends on the allowPing setting in rest_api category